build.ps1

```diff
@@ -0,0 +1,97 @@
+[CmdletBinding(PositionalBinding = $false)]
+Param(
+  [Parameter(ValueFromRemainingArguments = $true)][String[]]$verbs,
+  [Switch]$reset,
+  [Switch]$prereqs
+)
+
+function Get-Help() {
+  Write-Host "Actions (defaults to help):"
+  Write-Host "  init                      Performs the initalization of the repository, and installs/downloads"
+  Write-Host "                            all the required artifacts (i.e. driver), and initializes the submodule."
+  Write-Host "                            -reset     Forces the submodule to be updated."
+  Write-Host "                            -prereqs   Installs all the required prerequisites and dev certificates."
+  Write-Host ""
+  Write-Host "  roll [commit sha]         Moves the submodule to the commit specified, generates the API, and downloads"
+  Write-Host "                            downloads the new driver."
+  Write-Host "  driver                    Downloads the driver."
+  Write-Host "  help                      Prints this message."
+}
+
+function Invoke-Init() {
+  Invoke-InitializeSubmodule
+  if ($prereqs) { Invoke-InstallRequirements }
+  Invoke-DownloadDriver
+}
+
+function Get-SubmoduleStatus() {
+  $smStatus = git submodule status
+  return $smStatus
+}
+
+function Invoke-InitializeSubmodule($enableReset = $true) {
+  # Check the status of the submodule, if not initialized, let's do that
+  if ((Get-SubmoduleStatus).StartsWith("-") -or ($enableReset -and $reset)) {
+    Write-Host "🔨 Initializing git submodule..." -NoNewline
+    git submodule update --init >$null 2>&1
```

Comment:
FWIW, replacing this line with `git clone https://github.com/microsoft/playwright && git checkout $UPSTREAM_SHA` would be equal! However, both are fine.